src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -209,33 +223,39 @@ def handle_warning_click(self, button):
         self.data_submitted = "OK" in button.text()
 
     def reduce_dimensions(self, dim_number=None):
-        self.manual_reduction.clicked.disconnect()
-        self.auto_reduction.clicked.disconnect()
-        self.engine.reduce_dimensions(dim_number)
+        self.engine.state.imported_data.drop(self.reduced_columns, axis=1, inplace=True, errors='ignore')
+        self.engine.state.raw_data.drop(self.reduced_columns, axis=1, inplace=True)
+        self.reduced_columns = self.engine.reduce_dimensions(dim_number)
         self.get_data()
         self.show_reduction_results()
 
-    def render_reduction_group(self, max_dimensions):
-        for i in reversed(range(self.auto_reduction_group_layout.count())):
-            self.auto_reduction_group_layout.itemAt(i).widget().setParent(None)
-
-        self.num_dimensions_spinbox.setMinimum(2)
+    def set_reduction_bounds(self, max_dimensions):
```

Comment:
something is wrong with setting the bounds, sometimes after already making one reduction the value equals -1 and the program crashes when we run it with that number